pootle/apps/accounts/utils.py

```diff
@@ -242,30 +244,34 @@ def revert_units_edited(self):
         """
         stores = set()
         # Revert unit target where user is the last submitter.
-        for unit in self.user.submitted.iterator():
+        for change in self.user.units_submitted.iterator():
+            unit = change.unit
             stores.add(unit.store)
 
             # Find the last submission by different user that updated the
             # unit.target.
             edits = unit.get_edits().exclude(submitter=self.user)
-
+            user = None
+            submitted_on = None
             if edits.exists():
-                last_edit = edits.latest("pk")
+                last_edit = edits.order_by("-creation_time", "-pk").first()
                 unit.target_f = last_edit.new_value
-                unit.submitted_by_id = last_edit.submitter_id
-                unit.submitted_on = last_edit.creation_time
+                user = last_edit.submitter
+                unit.change.submitted_by_id = last_edit.submitter
+                unit.change.submitted_on = last_edit.creation_time
+                submitted_on = last_edit.creation_time
```

Comment:
If you invert the order you can use `submitted_on` instead of `last_edit.creation_time`:
```python
                submitted_on = last_edit.creation_time
                unit.change.submitted_on = submitted_on
```